clang/docs/LanguageExtensions.rst

```diff
@@ -434,6 +434,37 @@ __datasizeof
 ``__datasizeof`` behaves like ``sizeof``, except that it returns the size of the
 type ignoring tail padding.
 
+.. _builtin_structured_binding_size-doc:
+
+__builtin_structured_binding_size (C++)
+---------------------------------------
+``__builtin_structured_binding_size`` returns the *structured binding size*
+([dcl.struct.bind]) of the type ``T`` (or unevaluated expression ``arg``)
```

Comment:
I wonder how valid the expression form of this is.  The reason `sizeof` needs to take both is because C doesn't have `decltype`.  I quite dislike builtins that take expressions because they aren't as obvious that they are unevaluated, despite being in a `function call` syntax location.

Reply:
Yeah, to be perfectly honest, I went with that because we don't have non-boolean unary type traits.
But if people feel that we should not accept expressions, I can add a new Expr node to support them.